cases/src/org/commcare/cases/util/CasePurgeFilter.java

```diff
@@ -315,6 +324,23 @@ public boolean invalidEdgesWereRemoved() {
         return this.invalidEdgesWereRemoved;
     }
 
+    public String getMissingCasesString() {
+        return flattenVectorOfStrings(this.missingCases);
+    }
+
+    public String getRemovedCasesString() {
+        return flattenVectorOfStrings(this.casesRemovedDueToMissingCases);
+    }
+
+    private static String flattenVectorOfStrings(Vector<String> v) {
+        String s = "";
```

Comment:
For what it's worth, is good to get into the pattern of using a StringBuilder for this kind of thing, which is easier on the heap.